components/wpa_supplicant/src/utils/wpa_debug.c

```diff
@@ -50,14 +50,19 @@ int  wpa_snprintf_hex(char *buf, size_t buf_size, const u8 *data, size_t len)
 #ifdef DEBUG_PRINT
 void  wpa_dump_mem(char* desc, uint8_t *addr, uint16_t len)
 {
+    char output[40];
     wpa_printf(MSG_DEBUG, "%s\n", desc);
     if (addr){
         uint16_t i=0;
-        for (i=0; i<len; i++){
-            if (i%16==0) wpa_printf(MSG_DEBUG, "\n");
-            wpa_printf(MSG_DEBUG, "%02x ", addr[i]);
+        for (i = 0; i < len / 16; i++) {
+            wpa_snprintf_hex(output, 40, addr + i * 16, 16);
```

Comment:
Sure I can add a whitespace, the important part was not to have a newline for each byte.

Reply:
Realized there was a reason for why I didn't include the white space to start with. I reused the functions for hex strings that (at least in mainline wpa_supplicant) need them to be without white spaces due to non debug printing usage.

I haven't checked whether the functions wpa_snprintf_hex and wpa_snprintf_uppercase are linked and used by other non open source components of ESP-IDF so to avoid breakage of other functionality the implementation now supports both with and without white spaces.

Reply:
Realized there was a reason for why I didn't include the white space to start with. I reused the functions for hex strings that (at least in mainline wpa_supplicant) need them to be without white spaces due to non debug printing usage.

I haven't checked whether the functions wpa_snprintf_hex and wpa_snprintf_uppercase are linked and used by other non open source components of ESP-IDF so to avoid breakage of other functionality the implementation now supports both with and without white spaces.